### PR TITLE
Add test for unicode request param parsing

### PR DESCRIPTION
**Pull request by alexmadjar**
Shows the django doesn't have the unicode request param parsing bug.

### CODE REVIEW

File: tests/requests/tests.py [hunk 1]
```diff
@@ -557,6 +557,26 @@ def test_get_raw_uri(self):
...
         self.assertEqual(request.get_raw_uri(), 'http://evil.com/path/with:colons')
 
+    @override_settings(ALLOWED_HOSTS=['example.com'])
+    def test_get_params(self):
+        factory = RequestFactory(HTTP_HOST='evil.com')
+        request = factory.get('/?foo=bar')
+        self.assertTrue('foo' in request.GET.keys())
+        self.assertEqual(request.GET['foo'], 'bar')
+
+        request = factory.get(u'/?foo=☂')
```
Comment (ryochiji) [inline tests/requests/tests.py]: We should urlencode the value since it's otherwise not technically a valid request.
